File: backend/zvms/views/thoughts.py
from zvms.routelib import *
import zvms.impls.thoughts

route(
    rule='/thoughts',
    method='GET',
    impl_func=zvms.impls.thoughts.search_thoughts,
)

route(
    rule='/thoughts/<int:stuId>/<int:volId>',
    method='PATCH',
    impl_func=zvms.impls.thoughts.update_thought
)

File: apis/paths/thoughts.ts
import { $ } from "zvms-apis-paths-gen";
import { authData as a } from "../users/index.js";
import { int, structs } from "../types/index.js";

export default $(
    "/thoughts",
    [],
    {
        type: "GET",
        name: "search_thoughts"
    },
    $(
        "/<int:stuId>",
        [],
        {
            type: "GET",
            name: "signup",
            desc: "",
            req: {
                volid: int()
            }
        },
        $(
            "/<int:volId>",
            [],
            {
                type: "GET",
                name: "get_thought_info",
                auths: []
            },
            {
                type: "PATCH",
                name: "update_thought",
                auths: []
            }
        )
    )
);